### PR TITLE
Updated layout.

### PR DESCRIPTION
**Pull request by Elland**
This is not a full layout update. Trying to keep PRs shorter, so doing on a sort of screen by screen basis. Of course, sometimes some stuff leaks. I also got a lot of low hanging fruit with this one.

screen shot 2017-05-16 at 12 10 03
screen_shot_2017-05-16_at_10 39 04_720

### CODE REVIEW

**Comment by Elland** [comment]
Ready for review. 🚀

File: Token/Controllers/Messaging/ChatController.swift [hunk 1]
```diff
@@ -39,10 +39,11 @@ class ChatController: MessagesCollectionViewController {
 
     var textLayoutQueue = DispatchQueue(label: "com.tokenbrowser.token.layout", qos: DispatchQoS(qosClass: .default, relativePriority: 0))
 
-    lazy var rateButton: UIBarButtonItem = {
-        let view = UIBarButtonItem(title: "Rate", style: .plain, target: self, action: #selector(didTapRateUser))
+    lazy var avatarImageView: AvatarImageView = {
```
Comment (yuliaveres) [inline Token/Controllers/Messaging/ChatController.swift]: private

File: Token/Controllers/Settings/ProfileController.swift [hunk 2]
```diff
@@ -50,58 +48,84 @@ open class ProfileController: UIViewController {
...
 
-        view.layer.cornerRadius = 4.0
-        view.layer.borderColor = Theme.borderColor.cgColor
-        view.layer.borderWidth = Theme.borderHeight
-
         return view
     }()
 
-    lazy var aboutSeparatorView: UIView = {
+    lazy var editSeparatorView: UIView = {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private

File: Token/Controllers/Settings/ProfileController.swift [hunk 3]
```diff
@@ -50,58 +48,84 @@ open class ProfileController: UIViewController {
...
         let view = UIView(withAutoLayout: true)
         view.backgroundColor = Theme.borderColor
+        view.set(height: 1.0 / UIScreen.main.scale)
 
         return view
     }()
 
-    lazy var aboutTitleLabel: UILabel = {
+    lazy var aboutContentLabel: UILabel = {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private

File: Token/Controllers/Settings/ProfileController.swift [hunk 4]
```diff
@@ -50,58 +48,84 @@ open class ProfileController: UIViewController {
...
         let view = UILabel(withAutoLayout: true)
+        view.font = Theme.regular(size: 16)
+        view.textColor = Theme.lightGreyTextColor
+        view.numberOfLines = 0
 
         return view
     }()
 
-    lazy var locationSeparatorView: UIView = {
+    lazy var contentBackgroundView: UIView = {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private

File: Token/Controllers/Settings/ProfileController.swift [hunk 5]
```diff
@@ -50,58 +48,84 @@ open class ProfileController: UIViewController {
...
         return view
     }()
 
-    lazy var locationTitleLabel: UILabel = {
-        let view = UILabel(withAutoLayout: true)
-        view.text = "Location"
-        view.textColor = Theme.darkTextColor
-        view.textColor = .lightGray
-        view.font = .systemFont(ofSize: 15)
+    lazy var topSeparatorView: UIView = {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private

File: Token/Controllers/Settings/ProfileController.swift [hunk 6]
```diff
@@ -50,58 +48,84 @@ open class ProfileController: UIViewController {
...
+        view.layer.borderColor = Theme.borderColor.cgColor
+        view.layer.borderWidth = 1.0 / UIScreen.main.scale
+        view.set(height: 1.0 / UIScreen.main.scale)
 
         return view
     }()
 
-    lazy var locationContentLabel: UILabel = {
-        let view = UILabel(withAutoLayout: true)
+    lazy var reputationSeparatorView: UIView = {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: okay, all the rest .. 😄

File: Token/Controllers/Settings/ProfileController.swift [hunk 7]
```diff
@@ -155,87 +182,112 @@ open class ProfileController: UIViewController {
     }
 
     func addSubviewsAndConstraints() {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private ?

File: Token/Controllers/Settings/ProfileController.swift [hunk 8]
```diff
@@ -155,87 +182,112 @@ open class ProfileController: UIViewController {
...
-        self.view.addSubview(self.locationContentLabel)
+        self.view.addSubview(self.reputationSeparatorView)
+        self.view.addSubview(self.reputationView)
+        self.view.addSubview(self.bottomSeparatorView)
 
-        let height: CGFloat = 38.0
-        let marginHorizontal: CGFloat = 20.0
-        let marginVertical: CGFloat = 16.0
-        let avatarSize: CGFloat = 166
+        let height: CGFloat = 26.0
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: let keep this in constants ?

File: Token/Controllers/Settings/ProfileController.swift [hunk 9]
```diff
@@ -155,87 +182,112 @@ open class ProfileController: UIViewController {
...
-        self.avatarImageView.topAnchor.constraint(equalTo: self.topLayoutGuide.bottomAnchor, constant: 26).isActive = true
-        self.avatarImageView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true
+        self.avatarImageView.topAnchor.constraint(equalTo: self.topLayoutGuide.bottomAnchor, constant: 28).isActive = true
+        self.avatarImageView.leftAnchor.constraint(equalTo: self.view.leftAnchor, constant: marginHorizontal).isActive = true
 
         self.nameLabel.setContentHuggingPriority(UILayoutPriorityRequired, for: .vertical)
-        self.nameLabel.heightAnchor.constraint(greaterThanOrEqualToConstant: 24).isActive = true
-        self.nameLabel.topAnchor.constraint(equalTo: self.avatarImageView.bottomAnchor, constant: marginVertical).isActive = true
-        self.nameLabel.leftAnchor.constraint(equalTo: self.view.leftAnchor, constant: marginHorizontal).isActive = true
+        self.nameLabel.heightAnchor.constraint(greaterThanOrEqualToConstant: 22).isActive = true
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: all numeric values in constants, cause i have no idea where are those from, 
thus impossible to catch mistake if there is any

File: Token/Controllers/Settings/ProfileController.swift [hunk 10]
```diff
@@ -155,87 +182,112 @@ open class ProfileController: UIViewController {
...
+        self.bottomSeparatorView.rightAnchor.constraint(equalTo: self.view.rightAnchor).isActive = true
+
+        self.reputationBackgroundView.widthAnchor.constraint(equalTo: self.view.widthAnchor).isActive = true
+        self.reputationBackgroundView.topAnchor.constraint(equalTo: self.reputationSeparatorView.bottomAnchor).isActive = true
+        self.reputationBackgroundView.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
+        self.reputationBackgroundView.rightAnchor.constraint(equalTo: self.view.rightAnchor).isActive = true
+        self.reputationBackgroundView.bottomAnchor.constraint(equalTo: self.bottomSeparatorView.topAnchor).isActive = true
     }
 
+    func updateReputation() {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/ProfileController.swift]: private ?

File: Token/Theme/Theme.swift [hunk 11]
```diff
@@ -116,12 +104,22 @@ extension Theme {
...
+        return UIColor(hex: "EAEAEA")
+    }
+
+    public static var incomingMessageBackgroundColor: UIColor {
+        return .white
+    }
+
     public static var outgoingMessageTextColor: UIColor {
-        return self.lightTextColor
+        return UIColor.black
```
Comment (yuliaveres) [inline Token/Theme/Theme.swift]: Please, keep consistency
`UIColor.black`  vs  `.white`

File: Token/Theme/Theme.swift [hunk 12]
```diff
@@ -136,22 +134,22 @@ extension Theme {
...
     }
 
     static func bold(size: CGFloat) -> UIFont {
-        return UIFont(name: "SFUIDisplay-Bold", size: size) ?? UIFont.systemFont(ofSize: CGFloat(size), weight: UIFontWeightBold)
+        return UIFont(name: "SFUIText-Bold", size: size) ?? UIFont.systemFont(ofSize: CGFloat(size), weight: UIFontWeightBold)
     }
 
     static func medium(size: CGFloat) -> UIFont {
-        return UIFont(name: "SFUIDisplay-Medium", size: size) ?? UIFont.systemFont(ofSize: CGFloat(size), weight: UIFontWeightBold)
+        return UIFont(name: "SFUIText-Medium", size: size) ?? UIFont.systemFont(ofSize: CGFloat(size), weight: UIFontWeightBold)
```
Comment (yuliaveres) [inline Token/Theme/Theme.swift]: `weight: UIFontWeightBold...` ->>> `weight : UIFontWeightMedium` ?

File: Token/Views/MessageUI/MessageCell.swift [hunk 13]
```diff
@@ -18,7 +18,15 @@ import YYText
 
 class MessageCell: TGBaseMessageCell {
 
-    var bubbleImageView = UIImageView()
+    lazy var bubbleImageView: UIView = {
```
Comment (yuliaveres) [inline Token/Views/MessageUI/MessageCell.swift]: private / private(set) ?

Comment (yuliaveres) [inline Token/Views/MessageUI/MessageCell.swift]: for all ui elements

File: Token/Views/MessageUI/ImageMessageCell.swift [hunk 14]
```diff
@@ -55,8 +55,7 @@ class ImageMessageCell: MessageCell {
 
             self.bubbleView.frame = cellLayout.bubbleViewFrame
             self.bubbleImageView.frame = cellLayout.bubbleImageViewFrame
-            self.bubbleImageView.image = self.isHighlight ? cellLayout.highlightBubbleImage : cellLayout.bubbleImage
-            self.bubbleImageView.tintColor = cellLayout.isOutgoing ? Theme.outgoingMessageBackgroundColor : Theme.incomingMessageBackgroundColor
+            self.bubbleImageView.backgroundColor = cellLayout.isOutgoing ? Theme.outgoingMessageBackgroundColor : Theme.incomingMessageBackgroundColor
```
Comment (yuliaveres) [inline Token/Views/MessageUI/ImageMessageCell.swift]: can't u use `MessageCellLayout` here? (i am not sure, asking)  we have beautiful `Style` struct there, and all attributes and style properties might be returned directly from layout inctance, without any checks here? 🙏  ☺️

File: Token/Controllers/Messaging/ChatsFloatingHeaderView.swift [hunk 15]
```diff
@@ -42,17 +42,19 @@ class ChatsFloatingHeaderView: UIView {
         return button
     }
 
+    fileprivate var buttonAttributes: [String: Any] = [NSFontAttributeName: Theme.medium(size: 15), NSForegroundColorAttributeName: Theme.tintColor]
+
     lazy var requestButton: UIButton = {
```
Comment (yuliaveres) [inline Token/Controllers/Messaging/ChatsFloatingHeaderView.swift]: please, go over all ui elements in all classes you modified and check if we may hide or make them private(set) ? 
🥇

File: Token/Controllers/Contacts/FavoritesNavigationController.swift [hunk 16]
```diff
@@ -17,6 +17,9 @@ import UIKit
 
 public class FavoritesNavigationController: UINavigationController {
 
+    static let selectedContactKey = "Restoration::SelectedContact"
+
+
```
Comment (yuliaveres) [inline Token/Controllers/Contacts/FavoritesNavigationController.swift]: line ? 😃

File: Token/Controllers/Contacts/FavoritesController.swift [hunk 17]
```diff
@@ -281,6 +285,35 @@ open class FavoritesController: SweetTableController {
...
+            tabBarController.switch(to: .scanner)
+        }))
+
+        addContactSheet.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { action in
+            //
+        }))
+
+        addContactSheet.view.tintColor = Theme.tintColor
+        self.present(addContactSheet, animated: true, completion: {
+            // Due to a UIKit "bug", tint colour need be reset here.
```
Comment (yuliaveres) [inline Token/Controllers/Contacts/FavoritesController.swift]: 😕  o no , seriously ?

File: Token/Controllers/Apps/Controllers/BrowseNavigationController.swift [hunk 18]
```diff
@@ -39,5 +41,42 @@ public class BrowseNavigationController: UINavigationController {
...
     public override func viewDidLoad() {
         super.viewDidLoad()
+
+        // TODO: move restoration to the tabbar controller, so we only restore the currently selected.
+        if let appData = UserDefaults.standard.data(forKey: BrowseNavigationController.selectedAppKey) {
+            // we delay by one cycle and it's enough for UIKit to set the children viewcontrollers
+            // for the navigation controller. Otherwise `viewcontrollers` will be nil and it wont restore.
+            // Downside: it blinks the previous view for no.
+            // TODO: move all of this the the navigator so we can restore the hiararchy straight from the app delegate.
+            DispatchQueue.main.asyncAfter(seconds: 0.0) {
```
Comment (yuliaveres) [inline Token/Controllers/Apps/Controllers/BrowseNavigationController.swift]: why u need delay here?  `DispatchQueue.main.async`  doesn't do the job ? 😕

File: Token/Controllers/Contacts/FavoritesController.swift (hunk 17 above)
Comment (Elland) [inline Token/Controllers/Contacts/FavoritesController.swift]: Yes. 😢

File: Token/Controllers/Apps/Controllers/BrowseNavigationController.swift (hunk 18 above)
Comment (Elland) [inline Token/Controllers/Apps/Controllers/BrowseNavigationController.swift]: No. 😞 UIKit troubles.

**Comment by Elland** [comment]
Improvements are up!

**Comment by Elland** [comment]
@yuliaveres Moving on, we can go over changes again when you're back. 🚀

**Pull request merged**